### PR TITLE
Proper errors on Power and Product

### PR DESCRIPTION
**Pull request by kellertuer**
To continue #207 this PR introduces two special error types for composite manifolds like power and product: the `ComponentManifoldError` consists of an error and an index, where this error occurred,
the `CompositeManifoldError` collects a set of these if there is more than one error.

Maybe a separate docs page would be still nice.

### CODE REVIEW

File: docs/src/features/utilities.md [hunk 1]
```diff
@@ -14,3 +14,13 @@ Manifolds.StaticReshaper
 Manifolds.ArrayReshaper
 Manifolds.make_reshape
 ```
+
+# Specific error messages
```
Comment (mateuszbaran) [inline docs/src/features/utilities.md:18]: I would call this "Specific exception types".

File: test/groups/group_operation_action.jl [hunk 2]
```diff
@@ -14,6 +14,9 @@ include("group_utils.jl")
     @test repr(A_left) == "GroupOperationAction($(repr(G)), LeftAction())"
     @test repr(A_right) == "GroupOperationAction($(repr(G)), RightAction())"
 
+    @test switch_direction(LeftAction()) == RightAction()
+    switch_direction(RightAction()) == LeftAction()
```
Comment (mateuszbaran) [inline test/groups/group_operation_action.jl:18]:
```suggestion
    @test switch_direction(RightAction()) == LeftAction()
```

Comment (kellertuer) [reply test/groups/group_operation_action.jl:18]: Ah, yes. I am wondering by the way, why the group test coverage is so low, the tests here seem quite exhaustive, but then some do not seem to “hit” their lines.

Comment (mateuszbaran) [reply test/groups/group_operation_action.jl:18]: I don't know, I didn't check that in a long time. There are some real gaps in coverage there IIRC.

Comment (kellertuer) [reply test/groups/group_operation_action.jl:18]: Me neither, but at first glance, most things should be covered but aren't – so we might check somewhen, to get coverage a little further.

**Pull request merged**